vignettes/blueprintr.Rmd

```diff
@@ -40,88 +40,113 @@ cache_location <- tempdir()
+* Metadata, usually a `.csv` file that incorporates any mapping files and checks that need to be done on the dataset. 
 
-All blueprints have
+In order to create a blueprint, we use the `blueprint` function. This function takes three arguments: name (the name of your generated dataset), description (a description of your dataset), command (any functions that need to be applied in order to build the dataset). 
 
-* A name (the first argument) for the _target_ dataset.
-* A description or brief summary of what the target is. Can be `NULL`.
-* A command, which is a quoted statement that has the code for building this target.
-* A metadata location, which is a path to where the target metadata is saved.
+A project may need only a few blueprints, but more likely you'll need nested blueprints to transform the data. 
 
-<div class="vg-warning"><span>the blueprint name is "mtcars_dat" rather than "mtcars". If the two had the same name, drake would determine that the blueprint has a _circular dependency_ (it depends on itself). To avoid this, blueprints should not have the same names as global variables, like `mtcars`.</span></div>
+`blueprintr` has six "steps" (targets) per blueprint:
 
-To get this loaded into a drake plan, we need to _attach_ it to an already existing plan using
+Target name             | Description
+------------------------|--------------
+`{blueprint}_initial`   | The result of running the blueprint's `command`
+`{blueprint}_blueprint` | A copy of the blueprint to be used throughout the plan
+`{blueprint}_meta`      | A copy of the dataset metadata --- if the metadata file doesn't exist, it will be created in this step
+`{blueprint}_checks`    | Runs all tests on the `{blueprint}_initial` target
+`{blueprint}`           | The built dataset after running some cleanup tasks
+`{blueprint}_meta_path` | Creates the metadata file or loads it
 
-```r
-attach_blueprint(plan, blueprint)
-attach_blueprints(plan, ...)
-```
+<div class="vg-warning"><span>when writing other targets in your plan, it is advised to **not** refer to the `{blueprint}_initial` step since it could have problems which are discovered in the `{blueprint}_checks` step.</span></div>
 
-<div class="vg-info"><span>`attach_blueprints` accepts "[tidy dots](https://adv-r.hadley.nz/quasiquotation.html#tidy-dots)", so if you have a `list()` of blueprints, you can "splat" all of those blueprints into `attach_blueprints` like `attach_blueprints(plan, !!!list_of_blueprints)`</span></div>
 
-If you don't have an existing plan, you can create one with
+## Example
 
-```r
-plan_from_blueprint(blueprint)
-```
+Let's take a well known dataset-- `mtcars`, and create a blueprint for it.
 
-For now, we'll use an already existing plan, which is probably what you'll have most of the time.
+```{r}
+our_mtcars <- data.table::copy(mtcars)
+data.table::setDT(our_mtcars, keep.rownames = TRUE[])
```

Comment:
Why this call? `::copy()` does a deep clone of `our_mtcars` and converts it into a data.table. Unless it's for the rownames? In that case, I would create a column that's just for the rownames rather than keeping the rownames (they're an odd hack that's mainly beneficial to matrices, not dataframes).